app/service/interfaces/i_app_svc.py

```diff
@@ -20,6 +20,14 @@ def find_link(self, unique):
         """
         pass
 
+    @abc.abstractmethod
+    def find_op_with_link(self, link_id):
```

Comment:
I don't believe these need to all be abstract methods but it's consistent with the project